Flight/FlightManagement.cs

```diff
@@ -0,0 +1,275 @@
+                    filterPredicate = f => f.DepartureCountry.Equals(value);
+                    break;
+                case FilterParameter.DestinationCountry:
+                    filterPredicate = f => f.DestinationCountry.Equals(value);
+                    break;
+                case FilterParameter.DepartureDate:
+                    if (DateTime.TryParse(value, out DateTime departureDate))
+                        filterPredicate = f => f.DepartureDate.Date == departureDate.Date;
+                    else
+                    {
+                        Console.WriteLine("Invalid departure date value.");
+                        return;
+                    }
+                    break;
+                case FilterParameter.DepartureAirport:
+                    filterPredicate = f => f.DepartureAirport.Equals(value);
+                    break;
+                case FilterParameter.ArrivalAirport:
+                    filterPredicate = f => f.ArrivalAirport.Equals(value);
+                    break;
+                case FilterParameter.Class:
+                    if (Enum.TryParse(value, true, out FlightClass flightClass))
+                        filterPredicate = f => f.Class == flightClass;
+                    else
+                    {
+                        Console.WriteLine("Invalid class value.");
+                        return;
+                    }
+                    break;
+                default:
+                    Console.WriteLine("Invalid parameter.");
+                    return;
+            }
+
+            List<Flight> filteredFlights = Flights.Where(filterPredicate).ToList();
+
+            DisplayFilteredFlights(filteredFlights);
+        }
+
+        public void ListAllFlights()
+        {
+            try
+            {
+                Console.WriteLine("Listing all flights:\n");
+                Console.WriteLine("{0,-10} {1,-15} {2,-15} {3,-20} {4,-20} {5,-20} {6,-15} {7,-10}", "Flight", "Departure", "Destination", "Departure Date", "Departure Airport", "Arrival Airport", "Class", "Price");
+                Console.WriteLine(new string('-', 120));
+                if (Flights.Count == 0)
+                {
+                    throw new FlightManagementException("No flights found.");
+                }
+
+                DisplayFlights(Flights);
+
+                Console.WriteLine("\nDo you want to book a flight? (Y/N)");
+                string choice = Console.ReadLine();
+
+                if (choice.ToUpper() == "Y")
+                {
+                    BookFlight();
+                }
+                else
+                {
+                    Console.WriteLine("Thank you for using our flight booking service.");
+                }
+            }
+            catch (FlightManagementException ex)
+            {
+                Console.WriteLine($"An error occurred: {ex.Message}");
+            }
+        }
+
+        public void ViewPersonalBookings()
```

Comment:
the idea of this method to passing the user/passenger ID that I used at the beginning and filter the booked flights depend on the passengerId